### PR TITLE
feat: MFA method switching and enrollment guard

### LINKED ISSUE #294: Improve MFA login UX: method switching, button loading state
## Problem

When `mfa_method` is `both` and a user has enrolled in TOTP via account settings, login switches to TOTP with no way to fall back to email OTP. If the user doesn't have their authenticator app available, they're locked out.

## MFA method switching

Add a "Use a different method" link on the MFA verification page. Rules:

- Only show when `mfa_method` is `both` and SMTP is configured
- **Switch to email OTP**: always allowed (safe fallback for when authenticator app is unavailable)
- **Switch to TOTP**: only allowed if user already has `totp_verified = true`
- Never offer TOTP enrollment during login — enrollment belongs in account settings

When `mfa_method = both` and user has no TOTP enrolled, email OTP is the only method shown. No switcher, no enrollment prompt.

## TOTP enrollment

TOTP enrollment should only happen in the account UI, never during login. When `mfa_method = totp` only (no email fallback), enrollment during first login is the exception — there is no second channel available and the admin is making a conscious security tradeoff.

## Note

Button loading state was split into #307.

### PR DESCRIPTION
**Pull request by eugenioenko**
## Summary

Closes #294

- Adds "Use email instead" / "Use authenticator app instead" links on the MFA verification page when `mfa_method=both` and SMTP is configured, allowing users to switch between TOTP and email OTP during login
- Switching creates a new challenge (marks the old one as used) with the same expiration window — no time extension on switch
- Switching to TOTP is only allowed if the user has `totp_verified=true`; crafted URLs for unenrolled users are silently ignored
- TOTP enrollment during login is now restricted to `mfa_method=totp` only — when `mfa_method=both`, unenrolled users get email OTP instead of being forced through TOTP setup
- Adds defensive guard: if a TOTP challenge somehow reaches an unenrolled user under `mfa_method!=totp`, redirects to login instead of showing enrollment

## Test plan

- [x] 10 new unit tests covering all switching paths and edge cases
- [ ] Manual test: enable `mfa_method=both`, log in with TOTP-enrolled user → verify "Use email instead" link appears and works
- [ ] Manual test: log in with unenrolled user under `mfa_method=both` → verify email OTP is sent, no TOTP enrollment shown
- [ ] Manual test: on email OTP page, verify "Use authenticator app instead" link only appears for TOTP-enrolled users
- [ ] Manual test: verify switch link is hidden when `mfa_method=totp` or `mfa_method=email`

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: pkg/mfa/handler.go [hunk 1]
```diff
@@ -69,6 +69,59 @@ func handleMfaGet(w http.ResponseWriter, r *http.Request) {
 
 	cfg := config.Get()
 
+	// Method switching: handle switch_to query parameter
+	if switchTo := r.URL.Query().Get("switch_to"); switchTo != "" {
```
Comment (eugenioenko) [inline pkg/mfa/handler.go]: this probably needs to be moved to an internal helper function
